src/EditorFeatures/Core.Wpf/Suggestions/AsyncSuggestedActionsSource.cs

```diff
@@ -36,20 +38,40 @@ private partial class AsyncSuggestedActionsSource : SuggestedActionsSource, ISug
             {
             }
 
-            public async IAsyncEnumerable<SuggestedActionSet> GetSuggestedActionsAsync(
+            public async Task GetSuggestedActionsAsync(
                 ISuggestedActionCategorySet requestedActionCategories,
                 SnapshotSpan range,
-                [EnumeratorCancellation] CancellationToken cancellationToken)
+                ImmutableArray<ISuggestedActionSetCollector> collectors,
+                CancellationToken cancellationToken)
             {
                 AssertIsForeground();
+                try
+                {
+                    await GetSuggestedActionsWorkerAsync(
+                        requestedActionCategories, range, collectors, cancellationToken).ConfigureAwait(false);
+                }
+                finally
+                {
+                    // Always ensure that all the collectors are marked as complete so we don't hang the UI.
+                    foreach (var collector in collectors)
+                        collector.Complete();
```

Comment:
@AmadeusW i hope this is ok.  First, i really don't want to have to keep track of what collectors i already called Complete on.  Second, it feels weird that the callee needs to do this.  the caller can just move all collectors to the completed state once the async call has completed.

Reply:
That's not ok, calling Complete causes collector to send its SuggestedActionSets to be displayed. Note that when actions are displayed, they can't be reordered. The correct approach is to call .Add to add SuggestedActionSets for a given priority, and call .Complete when all actions for this priority are sent. This signals Editor to display these actions in the UI.